beta/nncf/tensorflow/sparsity/rb/algorithm.py

```diff
@@ -0,0 +1,222 @@
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import tensorflow as tf
+import tensorflow_addons as tfa
+from tensorflow.python.keras.utils.layer_utils import count_params
+
+from nncf.api.compression import CompressionScheduler
+from nncf.common.graph.transformations.commands import TransformationPriority
+from nncf.common.graph.transformations.layout import TransformationLayout
+from beta.nncf.tensorflow.algorithm_selector import TF_COMPRESSION_ALGORITHMS
+from beta.nncf.tensorflow.api.compression import TFCompressionAlgorithmBuilder
+from beta.nncf.tensorflow.graph.model_transformer import TFModelTransformer
+from beta.nncf.tensorflow.graph.transformations.commands import TFInsertionCommand
+from beta.nncf.tensorflow.graph.transformations.commands import TFLayerWeight
+from beta.nncf.tensorflow.graph.transformations.commands import TFOperationWithWeights
+from beta.nncf.tensorflow.graph.transformations.commands import TFRemovalCommand
+from beta.nncf.tensorflow.graph.transformations.layout import TFTransformationLayout
+from beta.nncf.tensorflow.graph.utils import collect_wrapped_layers
+from beta.nncf.tensorflow.graph.utils import get_original_name_and_instance_index
+from beta.nncf.tensorflow.graph.converter import convert_keras_model_to_nxmodel
+from beta.nncf.tensorflow.layers.wrapper import NNCFWrapper
+from beta.nncf.tensorflow.sparsity.base_algorithm import BaseSparsityController
+from beta.nncf.tensorflow.sparsity.rb.loss import SparseLoss, SparseLossForPerLayerSparsity
+from beta.nncf.tensorflow.sparsity.rb.operation import RBSparsifyingWeight, OP_NAME
+from beta.nncf.tensorflow.sparsity.rb.functions import binary_mask
+from beta.nncf.tensorflow.sparsity.schedulers import SPARSITY_SCHEDULERS
+from beta.nncf.tensorflow.sparsity.utils import convert_raw_to_printable
+from beta.nncf.tensorflow.utils.node import is_ignored
+
+
+PRUNING_LAYERS = {
+    'Conv1D': {'weight_attr_name': 'kernel'},
+    'Conv2D': {'weight_attr_name': 'kernel'},
+    'DepthwiseConv2D': {'weight_attr_name': 'depthwise_kernel'},
+    'Conv3D': {'weight_attr_name': 'kernel'},
+    'Conv2DTranspose': {'weight_attr_name': 'kernel'},
+    'Conv3DTranspose': {'weight_attr_name': 'kernel'},
+    'Dense': {'weight_attr_name': 'kernel'},
+    'SeparableConv1D': {'weight_attr_name': 'pointwise_kernel'},
+    'SeparableConv2D': {'weight_attr_name': 'pointwise_kernel'},
+    'Embedding': {'weight_attr_name': 'embeddings'},
+    'LocallyConnected1D': {'weight_attr_name': 'kernel'},
+    'LocallyConnected2D': {'weight_attr_name': 'kernel'}
+}
+
+
+@TF_COMPRESSION_ALGORITHMS.register('rb_sparsity')
+class RBSparsityBuilder(TFCompressionAlgorithmBuilder):
+    def __init__(self, config):
+        if isinstance(tf.distribute.get_strategy(), tf.distribute.MirroredStrategy):
+            raise Exception("RB sparsity algorithm do not support running in distributed mode")
```

Comment:
Please verify this catch. This doesn't catch multi gpu run in default distribution mode

Reply:
raise Exception("RB sparsity algorithm does not support the distributed mode with mirrored strategy")

Reply:
Please check it